frontend/src/metabase/common/components/EntityPicker/components/EntityPickerSearch/EntityPickerSearch.tsx

```diff
@@ -78,46 +75,45 @@ export function EntityPickerSearchInput<
 }
 
 export const EntityPickerSearchResults = <
-  Id,
+  Id extends SearchResultId,
   Model extends string,
   Item extends TypeWithModel<Id, Model>,
 >({
   searchResults,
   onItemSelect,
   selectedItem,
 }: {
-  searchResults: Item[] | null;
+  searchResults: SearchResult[] | null;
   onItemSelect: (item: Item) => void;
   selectedItem: Item | null;
 }) => {
-  const dispatch = useDispatch();
-
   if (!searchResults) {
     return <SearchLoadingSpinner />;
   }
 
   return (
-    <Box h="100%">
+    <Box h="100%" bg="bg-light">
       {searchResults.length > 0 ? (
         <Stack h="100%">
           <VirtualizedList
             Wrapper={({ children, ...props }) => (
-              <Box p="lg" {...props}>
-                {children}
+              <Box p="xl" {...props}>
+                <ChunkyList>{children}</ChunkyList>
               </Box>
             )}
           >
-            {searchResults?.map(item => (
-              <EntityPickerSearchResult
+            {searchResults?.map((item, index) => (
+              <ResultItem
                 key={item.model + item.id}
-                result={Search.wrapEntity(item, dispatch)}
-                onClick={(item: WrappedResult) => {
+                item={item}
+                onClick={() => {
```

Comment:
it looks like you can probably stop doing the type casting on the line below

Reply:
sadly no, the boundary between a SearchResult and an Item here is still a little messy